<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Kristian Hovde Liland, Hilde Vinje and Lars Snipen" />

<meta name="date" content="2020-08-11" />

<title>Introduction to microclass</title>

<script>// Hide empty <a> tag within highlighted CodeBlock for screen reader accessibility (see https://github.com/jgm/pandoc/issues/6352#issuecomment-626106786) -->
// v0.0.1
// Written by JooYoung Seo (jooyoung@psu.edu) and Atsushi Yasumoto on June 1st, 2020.

document.addEventListener('DOMContentLoaded', function() {
  const codeList = document.getElementsByClassName("sourceCode");
  for (var i = 0; i < codeList.length; i++) {
    var linkList = codeList[i].getElementsByTagName('a');
    for (var j = 0; j < linkList.length; j++) {
      if (linkList[j].innerHTML === "") {
        linkList[j].setAttribute('aria-hidden', 'true');
      }
    }
  }
});
</script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Introduction to microclass</h1>
<h4 class="author">Kristian Hovde Liland, Hilde Vinje and Lars Snipen</h4>
<h4 class="date">2020-08-11</h4>



<div id="k-mers" class="section level2">
<h2>K-mers</h2>
<p>All methods in this package are based on K-mer counting. K-mers are words of length <span class="math inline">\(K\)</span>, found in the DNA sequence being analysed. There are <span class="math inline">\(4^K\)</span> possible words of length <span class="math inline">\(K\)</span>, and all instances in a sequence are counted. Given the sequence “ATGCCTGAACTGACCTGC” we can, for instance, count 1-mers, 2-mers and 3-mers as follows:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">library</span>(microclass)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw">KmerCount</span>(<span class="st">&quot;ATGCCTGAACTGACCTGC&quot;</span>, <span class="dt">K =</span> <span class="dv">1</span>, <span class="dt">col.names =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<pre><code>##      A C G T
## [1,] 4 6 4 4</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="kw">KmerCount</span>(<span class="st">&quot;ATGCCTGAACTGACCTGC&quot;</span>, <span class="dt">K =</span> <span class="dv">2</span>, <span class="dt">col.names =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<pre><code>##      AA AC AG AT CA CC CG CT GA GC GG GT TA TC TG TT
## [1,]  1  2  0  1  0  2  0  3  2  2  0  0  0  0  4  0</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="kw">KmerCount</span>(<span class="st">&quot;ATGCCTGAACTGACCTGC&quot;</span>, <span class="dt">K =</span> <span class="dv">3</span>, <span class="dt">col.names =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<pre><code>##      AAA AAC AAG AAT ACA ACC ACG ACT AGA AGC AGG AGT ATA ATC ATG ATT CAA CAC
## [1,]   0   1   0   0   0   1   0   1   0   0   0   0   0   0   1   0   0   0
##      CAG CAT CCA CCC CCG CCT CGA CGC CGG CGT CTA CTC CTG CTT GAA GAC GAG GAT
## [1,]   0   0   0   0   0   2   0   0   0   0   0   0   3   0   1   1   0   0
##      GCA GCC GCG GCT GGA GGC GGG GGT GTA GTC GTG GTT TAA TAC TAG TAT TCA TCC
## [1,]   0   1   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0
##      TCG TCT TGA TGC TGG TGT TTA TTC TTG TTT
## [1,]   0   0   2   2   0   0   0   0   0   0</code></pre>
</div>
<div id="classification-in-one-line-of-code" class="section level2">
<h2>Classification in one line of code</h2>
<p>The quickest way to perform prokaryote classification is through the <code>taxMachine</code>. It is a genus level classification methods based on the multinomial method, trained on the <code>microcontax</code> data set.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="kw">library</span>(microcontax)</span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="kw">data</span>(small<span class="fl">.16</span>S)</span>
<span id="cb7-3"><a href="#cb7-3"></a>tax.tab &lt;-<span class="st"> </span><span class="kw">taxMachine</span>(small<span class="fl">.16</span>S<span class="op">$</span>Sequence)</span></code></pre></div>
<pre><code>## Welcome to the taxMachine!
##    classifying 71 sequences with median length 1338 ...
##    creating model...
##    caching model...
##    classifying sequence 1 to 71 
## done!</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="kw">head</span>(tax.tab)</span></code></pre></div>
<pre><code>##          Genus  D.score     R.score P.recognized
## 1 Caldisphaera 7.950911  0.05188227   1.00000000
## 2 Caldisphaera 8.692077 -0.07626455   0.18882257
## 3 Caldisphaera 8.653983  0.22779976   1.00000000
## 4 Caldisphaera 8.549670  0.23457126   1.00000000
## 5 Caldisphaera 8.580826  0.28430467   1.00000000
## 6 Caldisphaera 8.062746 -1.21785848   0.05899272</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>genus &lt;-<span class="st"> </span><span class="kw">sapply</span>(<span class="kw">strsplit</span>(small<span class="fl">.16</span>S<span class="op">$</span>Header,<span class="dt">split=</span><span class="st">&quot; &quot;</span>),<span class="cf">function</span>(x){x[<span class="dv">2</span>]})</span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="kw">cat</span>(<span class="st">&quot;Number of errors:&quot;</span>, <span class="kw">sum</span>(genus <span class="op">!=</span><span class="st"> </span>tax.tab<span class="op">$</span>Genus))</span></code></pre></div>
<pre><code>## Number of errors: 0</code></pre>
<p>Notice that a model is created and cached in memory the first time you call <code>taxMachine()</code> in a session. This is to avoid storing a huge matrix. The next call to <code>taxMachine()</code> in this session will be (much) faster, since it will re-use the created model.</p>
<div id="parallell-computing" class="section level3">
<h3>Parallell computing</h3>
<p>The number of processing cores to be used is controlled by the <code>setParallel()</code> command. The default is to used all available cores (logical and physical). Please remember to set this prior to a call if you run this on a computing cluster!</p>
</div>
</div>
<div id="uncertainty" class="section level2">
<h2>Uncertainty</h2>
<p>All input sequences are given a classification by the <code>taxMachine()</code>, but some may be very uncertain. The <code>taxMachine()</code> also calculates two types of uncertainties: the <code>D.score</code> and the <code>R.score</code>. A small <code>D.score</code> (close to 0, less than 1) indicates an uncertain result because the sequence is also very similar to another genus, i.e. its posterior probability is almost equal for two different genera. The <code>R.score</code> indicates if the sequence is very different from any previously seen sequence, and the more negative the <code>R.score</code>, the more ‘alien’ the sequence is to the <code>taxMachine</code>.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>), <span class="dt">mar =</span> <span class="kw">c</span>(<span class="fl">2.5</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">1</span>))</span>
<span id="cb13-2"><a href="#cb13-2"></a><span class="kw">boxplot</span>(tax.tab<span class="op">$</span>D.score, <span class="kw">factor</span>(genus), <span class="dt">main =</span> <span class="st">&quot;D-score&quot;</span>)</span>
<span id="cb13-3"><a href="#cb13-3"></a><span class="kw">boxplot</span>(tax.tab<span class="op">$</span>R.score, <span class="kw">factor</span>(genus), <span class="dt">main =</span> <span class="st">&quot;R-score&quot;</span>)</span></code></pre></div>
<div class="figure">
<img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAASAAAAEgCAMAAAAjXV6yAAAAn1BMVEUAAAAAADoAAGYAOjoAOmYAOpAAZrY6AAA6ADo6OgA6OmY6OpA6ZpA6ZrY6kLY6kNtmAABmOgBmOjpmkLZmkNtmtttmtv+QOgCQZgCQZjqQkGaQtpCQttuQtv+Q2/+2ZgC2Zjq2kDq229u22/+2///T09PbkDrbkGbbtmbbtpDb27bb29vb2//b/9vb////tmb/25D/27b//7b//9v///+3XE6QAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAH9ElEQVR4nO2daXubRhSFsWPZjZPUcptYXdKSxi5N21Qr//+3dWYAgeUMZ1YYwTkfbB50752jVwMMe1ZSvcrGNpC6CAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICMgOUJEpXbx9jGTHUodVbeg2miEnQMLRQyQ/dmoACUO/R2rCEVA8Q1ZqAWXXkZqwBXT5JP79c9819NfrdqH7IqZffVCTf/8gJn9cq6yLX++zy8fy8ElEvluH8q4ALWS57U31P4IfJ0DlftkaqnuV+iCvpu/a2TKsqKdEltTVk1WbfWoAiYZj+XED1E5IVtfr8otyscmy79e7pfys+k0LNVv8XTyW/wq74lcTH9xZtdmnb/Sg0H6cATUrIdHA1ef641x9XLz93ExW/4Shh8r6XbdGAHXWQc23DO3HA5DqoYv/lEe1cB87fDupQNY0hfXQK/gWkOg1cfx4A1rv7qtWfpMumlW37Ojy/0b+WLWhTfgtYAPo6mPVZgQ/boBE05dPtaGy3P3yWk3hXyz06Ek1c/ijshHHjxOg3ep03HH4SX5SLeib24+ny7wy1PyKAVV/7+LFMCicH++BotpSlKJfC6di+p383cQv83yrUUXn2cWHctcZIXir6Rh5duwMof3472p86hDL21Hts3FHZagedwQchHc288dNUWA/AXZW5Wg1u+2MXN+rSTVyfV9l1R52P4sNR8gd3c5AsV1cwvrh4Q4gAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIyBJTFkbvtwfyYAnL+KnGqDueHgEBVAgJVEwa06TnpOXdAeZbdbd/IS34011/MHFC+WJe56j2F5sKC1AEF2l5r0lS/2X6nLnJ6dnmcc4OG2/fwPcjzh9QCktcOHr6WwXpQ9me/zgxQWbSX0GquQJ05oLKoNl8b7TVycwcUOnM0QJ4ioGhVCShsJtdBKJyAQDgBgXACAuEEBMKHAuRyUBdXdVCygMwaMvZtF+6eSUAonIBAOAGBcAIC4b6A5N1E6v5PcPxlroDkgc1c3nCFjuDNFNBhJW/CNzmLMFNA6ixCdf4AnEUYClDRLO2F5qb/mfcgeUh6v1R3miYBqF0HKVQ9hoYBVLk4rORtyoEA+Z4XS2sr1pxwzhfr4sxOHA7Zg4Ty61R6EG6h/jcIoOOCtV/qnswy95F0US/rhxUBuYmAwmYSEAonIBBOQCCcgEA4AYFwAgLhBATCCQiEDwbIbKfP2LdduHsmAaFwAgLhgwEya8jYt124eyYBoXACAuEEBMIJCIQTEAgnIBA+LUD1U3qVeND+W9KdzNCOSNGA1XCAezaABCHwrHBNpvsyq01PE1DPrXS9mfMB5JhJQCCTgPwzbasSEKhKQKDqtABxHQQyCejUT/xDwNW8vHpFGXxeT3KA4vtR8xQf+bQV9LyemQJSNyDkau8njeukjTMHA/TQXFmayJX2ppmDLWKi9xTsQXpA++XlU+d+hFENWWUOBej4vj/tUZjZA0rHkJ2focZBxoYICBgiIGAoNUDx/RAQ8JMmoOYNvNq3MM8cUNG82nqje5P3vAF1bgwdf2SfIqDOMYXx9w1TBMQehBpvbjjkOkj3QXN9h/Y11XMHZOyHI2m7TPeC2nQCAukEBNIJCKRPC5CnCMihKgGBqtMCxHUQyBwfUPyRq1fm6IACa/qAIhjyyiSgQA1bVCUgUJWAQNVpAUpuHeQpTVX3+8VmAkj/8DuUORdA53u/mGO2ZdXyfO8Xc8y2rKqPd92n8fBzVoCCNWxRtTtve6Pt12MA0j672b9hi6pyXlKb1VYOgGKtg+ozvPY9yFNnA0j0IXmS9wWgMVaKrVICVKpXK7MH9c0rsrvEVtIWmZ5HNvv8tPO2N6/OFlAggc38YaUfvhKQQ6p5tmXVqJm2VQkIVE1zbz5ipm1VAgJVCQhU5ToIVCUgUJWAQFUCAlXdAEXc94mdaVuVWzFQlYBAVQICVQkIVCUgUPWsAJ3fiUNtuNnFnqF7kC+4AQGZXS4cGJB315o4IAXHi9DUAXnWJCCPqk6pfeFcxED4GIC4koZ+uJl39eNRlYBAVQICVQkIVCUgUHVSgLgV6/eTeZXUJE8IEHc1gB8CAn5iAirE6k1dfleM/j6vVhv8jP2XM+OsgwphZL+UdyAlAijPsrvtmzV8xv7p3EhbseqC5MNqsQ4GyO+YdC6c5Kr3gKfgBZYOUPM7CV9F/3MLjVvyAqT8qLc0nDxHsf+r+Ku/B5Xy2fZJ9CD1loby8LVMpQcdF6z9cvyXLnb91Kj8/RiqbytWLWTau2kHXkkX1ebr9MZM50Xe1LbhPPewY3iIzXxAPx5VEwaUzh2Hjql94QQEwgkIhBMQCCcgEB4CUEA/HlUJCFQlIFCVgEBVAgJVCQhUJSBQlYBA1UkB4pnVfj+ZV0lN8oQAZZ41BwXkd1bDzc85ATIUAUWreu7rIEPNdytmqPmOgww1JUBx5P5VBvMTfqH1ifNVBD8EFC7UI5uAorZiLgIK1A4BBYqzC/XIJqCorZiLgAK1Q0CB4uxC5ykCAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAiIgIAICIiAgAgIiICACAvofbpDU5tbZpMgAAAAASUVORK5CYII=" alt />
<p class="caption">Classification uncertainty scores</p>
</div>
<p>Along with each <code>R.score</code> value is also a <code>P.recognized</code> value. This is simply the probability that one of the training data sequences get an <code>R.score</code> of the same value, or lower. It is perhaps easier to interpret than the score itself. We typically use this to mark some sequences as unrecognized, and risky to include in further analyses:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>tax.tab<span class="op">$</span>Is.Recognized &lt;-<span class="st"> </span>tax.tab<span class="op">$</span>P.recognized<span class="op">&gt;</span><span class="fl">0.01</span></span>
<span id="cb14-2"><a href="#cb14-2"></a>tax.tab[<span class="dv">35</span><span class="op">:</span><span class="dv">45</span>,]</span></code></pre></div>
<pre><code>##        Genus  D.score     R.score P.recognized Is.Recognized
## 35 Acidianus 5.425759  0.59778317   1.00000000          TRUE
## 36 Acidianus 7.367201  0.14176582   1.00000000          TRUE
## 37 Acidianus 5.464705  0.56362329   1.00000000          TRUE
## 38 Acidianus 7.879450 -0.35603991   0.13953400          TRUE
## 39 Acidianus 7.195179  0.09568198   1.00000000          TRUE
## 40 Acidianus 7.832235 -0.44416463   0.12710836          TRUE
## 41 Acidianus 9.064079 -3.69599537   0.00610501         FALSE
## 42 Acidianus 8.097803 -0.93641982   0.07726423          TRUE
## 43 Acidianus 7.438815  0.19891170   1.00000000          TRUE
## 44 Acidianus 7.438897  0.17761285   1.00000000          TRUE
## 45 Acidianus 7.394431  0.26012677   1.00000000          TRUE</code></pre>
<p>In most cases we would accept as recognized any sequence with <code>P.recognized&gt;1e-4</code>, since there is a substantial variation also in ‘proper’ sequences. Sequences with a lot of sequencing errors, chimera or completely new taxa usually get extremely negative <code>R.scores</code>.</p>
</div>
<div id="custom-classification" class="section level2">
<h2>Custom classification</h2>
<p>If classification is to be carried out on a different taxonomic level than genus, using a different training data set, or using different parameters, the RDP and multinomial methods are available.</p>
<p>The following example uses half the <code>small.16</code> data set to train an RDP model and then tries to classify the other half of the data.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a>rdp &lt;-<span class="st"> </span><span class="kw">rdpTrain</span>(small<span class="fl">.16</span>S<span class="op">$</span>Sequence[<span class="kw">seq</span>(<span class="dv">1</span>,<span class="dv">71</span>,<span class="dv">2</span>)], genus[<span class="kw">seq</span>(<span class="dv">1</span>,<span class="dv">71</span>,<span class="dv">2</span>)])  <span class="co"># training step</span></span>
<span id="cb16-2"><a href="#cb16-2"></a>predicted &lt;-<span class="st"> </span><span class="kw">rdpClassify</span>(small<span class="fl">.16</span>S<span class="op">$</span>Sequence[<span class="kw">seq</span>(<span class="dv">2</span>,<span class="dv">71</span>,<span class="dv">2</span>)], rdp)        <span class="co"># classification step</span></span>
<span id="cb16-3"><a href="#cb16-3"></a><span class="kw">cat</span>( <span class="st">&quot;Number of errors:&quot;</span>, <span class="kw">sum</span>(predicted <span class="op">!=</span><span class="st"> </span>genus[<span class="kw">seq</span>(<span class="dv">2</span>,<span class="dv">71</span>,<span class="dv">2</span>)]) )</span></code></pre></div>
<pre><code>## Number of errors: 0</code></pre>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
